modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/management/SchemaManager.java

```diff
@@ -451,6 +454,9 @@ public void onCacheDestroyed(String cacheName, boolean rmvIdx, boolean clearIdx)
             if (schema.decrementUsageCount()) {
                 schemas.remove(schemaName);
 
+                if (destroy)
```

Comment:
We just removed local schema. Why do we keep cluster schema if cache was only stopped? Also `if (!U.isLocalNodeCoordinator(ctx.discovery()))` might be moved outside.